### ISSUE TITLE
Does this convert from simple-schema to json-schema?

### ISSUE THREAD

**Issue by zeroasterisk** [clinical-meteor/schema-hydrator#2]
If so, this is MUCH more useful tool than just storing into a DB....

**Comment by awatson1978** [comment]
Not quite, but it's certainly a step in that direction.  As you know, class objects in javascript are simply JSON objects with functions.  So, the problem with storing SimpleSchemas to a database is that the `type` field expects a javascript class.  The `dehydrate()` function was intended to remove all of the functions from the schema object, so it could just be plain JSON.  

Beyond that, it doesn't rearrange the schema structure.  However, the SimpleSchema is pretty close to JSON-schema; so it wouldn't be too hard to do some restructuring and output to a new schema format.  

Although, to be perfectly honest, I'm looking at this code, and am rather skeptical that it will actually dehydrate as expected.
https://github.com/clinical-meteor/schema-hydrator/blob/master/lib/SchemaHydrator.js#L132-151

As I recall, we were having inversion problems in the data pipeline on the project where this package came from.   It would start off as one schema, then it would get inverted, and we would have to double invert to get back to the original.  

`(a,b,c) > !(!c' !b', !a) > (a,b,c)`

If you're going to dig around with this code, start by confirming that my last commit wasn't during the inversion step and things aren't precisely backwards.  Looking at the code, I think the following is the correct syntax (we should be replacing objects with strings, and not the other way around).

``` js
dehydrate: function (schema){
    if (process.env.DEBUG) {
      console.log('dehydrating schema...', schema);
    }

    Object.keys(schema).forEach(function (key){
      // console.log('hydrating schema key ', key);
      // console.log('hydrating schema key.type ', schema[key].type);
      if (schema[key].type === String) {
        schema[key].type = "String";
      }
      if (schema[key].type === Number) {
        schema[key].type = "Number";
      }
    });
    if (process.env.DEBUG) {
      console.log('dehydrated schema: ', schema);
    }
    return schema;
  }
```

**Comment by awatson1978** [comment]
I think you're looking for a function like this?  Where you can edit and restructure a simpleschema to be json-schema format?

``` js
convertToJsonSchema: function (simpleSchema){
  let jsonSchema = {};

    Object.keys(simpleSchema).forEach(function (key){

      jsonSchema[key] = simpleSchema[key];

      if (jsonSchema[key].type === String) {
        jsonSchema[key].type = "String";
      }
      if (jsonSchema[key].type === Number) {
        jsonSchema[key].type = "Number";
      }
    });

    return jsonSchema;
  }
```

Happy to take a pull request if you think this is needed functionality.

Also, relevant libraries which I've been thinking of adding to the Clinical Meteor track:
https://github.com/joelvh/json2json
http://jsoniq.org/
